libr/io/p/io_w32dbg.c

```diff
@@ -180,15 +180,14 @@ static char *__system(RIO *io, RIODesc *fd, const char *cmd) {
 	} else if (!strncmp (cmd, "pid", 3)) {
 		if (cmd[3] == ' ') {
 			int pid = atoi (cmd + 3);
-			if (pid > 0 && pid != iop->pi.dwProcessId) {
-				iop->pi.hProcess = OpenProcess (PROCESS_ALL_ACCESS, false, pid);
-				if (iop->pi.hProcess) {
-					iop->pi.dwProcessId = iop->pi.dwThreadId = pid;
+			if (pid > 0 && pid != iop->pi.dwThreadId && pid != iop->pi.dwProcessId) {
+				iop->pi.hThread = OpenThread (PROCESS_ALL_ACCESS, FALSE, pid);
+				if (iop->pi.hThread) {
+					iop->pi.dwThreadId = pid;
```

Comment:
don't manually change the `iop->pi.dwThreadId`. w32_continue needs it to always be the last tid that sent an event.

Reply:
In that case, wouldn't it be wise to also change w32_continue to work with the tid it is given? Otherwise w32_continue will always continue from whichever thread dbg_wait sets globally even though someone called `dc` after switching threads. `r_debug_native_continue` and `r_debug_continue_kill` don't seem to pay any attention to this detail.

Reply:
When calling `ContinueDebugEvent`, `dwThreadId` must be the one from the last debug event received. It doesn't mean the next debug event will be from this thread.

Calling `ResumeThread` should be enough to select and signal a thread to continue , thats what `w32_select` does, and `SuspendThread` is also called in all other threads if `dbg.threads = true`

Reply:
yep. it must populate that "current pid" to r2 itself

Reply:
As in `return r_str_newf ("%d", pid);`?